imagefit/conf.py

```diff
@@ -35,8 +41,12 @@ class ImagefitConf(AppConf):
     }
 
     #: ConditionalGetMiddleware is required for browser caching
-    if not 'django.middleware.http.ConditionalGetMiddleware' in settings.MIDDLEWARE_CLASSES:
-        settings.MIDDLEWARE_CLASSES += ('django.middleware.http.ConditionalGetMiddleware',)
+    if _USE_OLD_DJANGO_BEHAVIOR:
+        _middlewares_list = getattr(settings, 'MIDDLEWARE')
+    else:
+        _middlewares_list = getattr(settings, 'MIDDLEWARE_CLASSES')
+    if not 'django.middleware.http.ConditionalGetMiddleware' in _middlewares_list:
+        _middlewares_list += ('django.middleware.http.ConditionalGetMiddleware',)
```

Comment:
Instead of sniffing the Django version, why not detecting which feature is active? Something like: (untested code)

```python
property_name = (hasattr(settings, 'MIDDLEWARE') and 'MIDDLEWARE' or 'MIDDLEWARE_CLASSES'
```

Also, I don't think that `_middlewares_list` will be pushed into the `settings.MIDDLEWARE` unless doing it explicitely.

```python
middleswares = getattr(settings, property_name)
if not 'django.middleware.http.ConditionalGetMiddleware' in middlewares:
    setattr(settings, property_name, middlewares + ('django.middleware.http.ConditionalGetMiddleware',))
```

Again, this is untested and typed in the Github textarea, so please consider this as just an explicit though ;)

Does this make sense?

Reply:
> Instead of sniffing the Django version, why not detecting which feature is active? Something like: (untested code)

Good idea, I'll do that.

> Also, I don't think that _middlewares_list will be pushed into the settings.MIDDLEWARE unless doing it explicitely.

I think `_middlewares_list` is a reference to the list. I just tried this in a test project without this middleware with a view containing a `print(settings.MIDDLEWARE)` and 'django.middleware.http.ConditionalGetMiddleware' appeared in the list.

Both solutions should work fine, so which one do you prefer? :)

Reply:
That's a good point to know that it works as a reference.
I would feel more comfortable with and explicit injection as if it was obvious, we wouldn't be discussing about this ;)

Reply:
Indeed, you're right.

I just pushed the changes. Tell me if it's ok and I'll do a squash. ;)